### PR TITLE
slimerjs: adapt requirement for simplicity

### PR DESCRIPTION
**Pull request by DomT4**
Re #36635

### CODE REVIEW

**Comment by DomT4** [comment]
@mikemcquaid

File: Library/Formula/slimerjs.rb [hunk 1]
```diff
@@ -1,5 +1,6 @@
 class FirefoxDependency < Requirement
   fatal true
+  default_formula "xulrunner"
```
Comment (mikemcquaid) [inline Library/Formula/slimerjs.rb]: Probably wants a Yosemite check on this line.

File: Library/Formula/slimerjs.rb [hunk 2]
```diff
@@ -7,10 +8,10 @@ def self.firefox_installation
     paths.find { |p| File.exist? File.expand_path(p) }
   end
 
-  satisfy { FirefoxDependency.firefox_installation }
+  satisfy { FirefoxDependency.firefox_installation || Formula["xulrunner"].installed? }
```
Comment (mikemcquaid) [inline Library/Formula/slimerjs.rb]: I'd reverse the order.

File: Library/Formula/slimerjs.rb (hunk 1 above)
Comment (DomT4) [inline Library/Formula/slimerjs.rb]: There's already a Yosemite failure in the formula for ` xulrunner ` - I didn't know if that would be sufficient?

File: Library/Formula/slimerjs.rb (hunk 2 above)
Comment (DomT4) [inline Library/Formula/slimerjs.rb]: I didn't know if we wanted to always prefer Firefox if it was available, or always prefer our ` xulrunner ` formula if available. Will flip them around :+1:

File: Library/Formula/slimerjs.rb (hunk 1 above)
Comment (mikemcquaid) [inline Library/Formula/slimerjs.rb]: It'll probably try and install it and then fail in that case.

Comment (DomT4) [inline Library/Formula/slimerjs.rb]: Alright, I'll stick a check in place in the requirement here.

Comment (DomT4) [inline Library/Formula/slimerjs.rb]: Pushed a check. Don't know if the check shells out, which I know Jack has been concerned about in requirements before, but I guess it matters less in a formula? Let me know if we need to get fancier on the check though.

**Comment by DomT4** [comment]
Thanks for all the help getting this done @mikemcquaid  & @skycaptain. Seriously appreciated :+1:.

**Comment by skycaptain** [comment]
You're welcome and thank you, too, since you had all the work :smile:

**Pull request closed without merging**